hash/bucketer.go

```diff
@@ -86,36 +86,44 @@ func (b *Bucket) Has(nn types.NamespacedName) bool {
 	return b.buckets.Owner(nn.String()) == b.name
 }
 
+// NewBucket creates a new Bucket with given name based on this bucketset.
+func (bs *BucketSet) NewBucket(name string) *Bucket {
+	return &Bucket{
+		name:    name,
```

Comment:
Since you're here, I think it's chance to add a check that the given name is in the bucketSet.buckets.

Reply:
+1

Reply:
Sounds good!
Done.